File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.ts
function maxScore(s: string): number {
    let n = s.length
    let ones_count = 0

    for(let i=0; i < n; i++){
        if(s[i] === "1"){
            ones_count++
        }
    }
    let zeroes_count = 0;
    let max_score = 0;

    for(let j=0; j < n-1; j++){
        if(s[j]==="1"){
            ones_count--
        }else{
            zeroes_count++
        }
        if((ones_count + zeroes_count) > max_score){
            max_score = ones_count + zeroes_count
        }
    }
    return max_score
};

File: 1422-maximum-score-after-splitting-a-string/1422-maximum-score-after-splitting-a-string.py
class Solution:
    def maxScore(self, s: str) -> int:
        n: int = len(s)
        ones_count: int = 0
        zeroes_count: int = 0
        max_score: int = 0

        for i in range(n):
            if s[i] == '1':
                ones_count += 1
                
        for i in range(n-1):
            if s[i] == '1':
                ones_count -= 1
            else:
                zeroes_count += 1
                
            max_score = max(max_score, ones_count + zeroes_count)
        
        return max_score      